package/src/components/MessageList/MessageList.tsx

```diff
@@ -834,13 +843,15 @@ const MessageListWithContext = <
     scrollToDebounceTimeoutRef.current = setTimeout(() => {
       // goToMessage method might have requested to scroll to a message
       let messageIdToScroll: string | undefined = messageIdToScrollToRef.current;
-      if (!initialScrollSet.current && initialScrollToFirstUnreadMessage) {
+      const countUnread = channelRef.current?.countUnread();
+      if (
+        !initialScrollSet.current &&
+        initialScrollToFirstUnreadMessage &&
+        countUnread > scrollToFirstUnreadThreshold
+      ) {
         // find the first unread message, if we have to initially scroll to an unread message
-        for (let index = messageList.length - 1; index >= 0; index--) {
-          if (isUnreadMessageRef.current(messageList[index], channelLastReadRef.current)) {
-            messageIdToScroll = messageList[index].id;
-            break;
-          }
+        if (messageList.length > countUnread) {
+          messageIdToScroll = messageList[countUnread - 1].id;
```

Comment:
Shouldn't this be:

```
messageIdToScroll = messageList[messageList.length - countUnread - 1].id;
```
 Let's say a channel has 100 messages and 10 latest messages are unread; then the index will be (100-10-1)=89, which will be from the end.
In your case, it will be (10-1)=9, which will then point from the top.

Generally, the scroll will happen from bottom to top if I am not wrong.

I could be wrong, just want to understand it.

Reply:
This is also how I understand the current state of affairs, how @khushal87 is explaining it

Reply:
actually this should have been ` if (messageList.length >= countUnread) {`

rest of it is fine. 

The way it works is like this.. say 10th message from bottom is unread.. we load 25 messages around it

so message list will 25 + 10 messages + (no 15 messages after) = 35 messages -- the messageList length.

Reply:
And this is only when the messageSet is the latest.. which is always the case when scrollToFirstUnread is true

in other cases we cannot use countUnread..

Thats why I use this guard when checking to show the unread indicatior

```
const isLatestMessageSetShown = channel.state.messageSets.find(
          (set) => set.isCurrent && set.isLatest,
        );
 return isLatestMessageSetShown && messageArrayIndex <= countUnread - 1;
```